src/components/Api.js

```diff
@@ -0,0 +1,135 @@
+class Api {
+  constructor({ baseURL, authToken, headers }) {
+    this._baseUrl = baseURL;
+    this._authToken = authToken;
+    this._headers = headers;
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`).then(res => res);
+  }
+
+  //methods for working with the API 
+  //GET https://around-api.en.tripleten-services.com/v1/cards
+  getCardList() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: {
+        authorization: this._authToken
+      }
+    }).then(res => {
+      res.ok ? res.json() : Promise.reject(`Error: ${this._handleServerResponse}`)
+    }).catch((err) => {
+      console.log(err); // log error to console
+    });
+  }
+
+  //GET https://around-api.en.tripleten-services.com/v1/users/me
+  getUserInfo(){
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: {
+      authorization: this._authToken
+      }
+    })
+    .then(res => {res.ok ? res.json() : Promise.reject(`Error: ${this._handleServerResponse}`)})
+    .catch((err) => {
+        console.log(err); //log error to console
+    });
+  }
+  getAppInfo() {
+    return Promise.all({this:getCardList(), this:getUserInfo()});
+  }
+
+  //POST https://around-api.en.tripleten-services.com/v1/cards
+  addCard({name, link}) {
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "POST",
+      headers: {
+      authorization: this._authToken,
+      "Content-Type": "application/json"
+      },
```

Comment:
According to the task you need to pass full `headers` into the constructor of `Api`. See here https://snipboard.io/pRdaAZ.jpg . And in the code you need to use only: `headers: this._headers` (not to duplicate the code)